pkg/core/blockchain.go

```diff
@@ -740,6 +740,15 @@ func (bc *Blockchain) GetAccountState(scriptHash util.Uint160) *AccountState {
 	return as
 }
 
+// GetUnspentCoinState returns unspent coin state for given tx hash.
+func (bc *Blockchain) GetUnspentCoinState(hash util.Uint256) *UnspentCoinState {
+	ucs, err := getUnspentCoinStateFromStore(bc.memStore, hash)
+	if err != nil {
+		ucs, _ = getUnspentCoinStateFromStore(bc.Store, hash)
```

Comment:
hm. didn't get an idea here.
First -- you check coinState in cache(memStore) and in case it's there you overwrite `ucs` with value from Store...
Why is that so?

Reply:
Found more obvious(as for me) logic in `GetStorageItem`:
`
sItem := getStorageItemFromStore(bc.memStore, scripthash, key)
	if sItem == nil {
		sItem = getStorageItemFromStore(bc.Store, scripthash, key)
	}
`

Reply:
> in case it's there

Nope, if there is an error we reach to the persistent store. It differs from `getStorageItemFromStore` because that one doesn't return any error.

Reply:
ok I see, ty!